File: week02/Longest_Common_Prefix.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        string result = strs[0];
        for (string str : strs)
        {
            for (int i = 0; i < result.size(); i++)
            {
                if (str[i] != result[i])
                    result = result.substr(0, i);
            }
        }
        return result;
    }
};

File: week02/longestCommonPrefix.js
var longestCommonPrefix = function (strs) {
  let commonPrefix = '';
  for (let i = 0; i < strs[0].length; i++) {
    if (strs.every((str) => str[i] === strs[0][i])) {
      commonPrefix += strs[0][i];
      continue;
    }
    break;
  }
  return commonPrefix;
};
